core/handler/gear_error.ex

```diff
@@ -84,9 +84,42 @@ defmodule AntikytheraCore.Handler.GearError do
          ) :: :ok do
     %Request{method: method, path_info: path_info} = request
     path = "/" <> Enum.join(path_info, "/")
-    gear_logger_module = CoreConn.gear_name(conn) |> GearModule.logger()
+    gear_name = CoreConn.gear_name(conn)
     log_message = "#{Method.to_string(method)} #{path} #{ErrorReason.format(reason, stacktrace)}"
-    gear_logger_module.error(log_message)
+
+    do_output_error_to_log(gear_name, log_message, reason)
+  end
+
+  if Antikythera.Env.compile_env() == :undefined do
+    alias AntikytheraCore.GearLog.Writer
```

Comment:
We'll get the following warning when setting `ANTIKYTHERA_COMPILE_ENV`:

```
$ ANTIKYTHERA_COMPILE_ENV=local mix compile
Compiling 202 files (.ex)
warning: unused alias Writer
  core/handler/gear_error.ex:94

Compilation failed due to warnings while using the --warnings-as-errors option
```

According to https://github.com/elixir-lang/elixir/issues/7980, we need to set `warn: false` to suppress the warning.

In addition, I think `Mix.env() == :test` is more intuitive than `Antikythera.Env.compile_env() == :undefined` for this case.

```suggestion
  if Mix.env() == :test do
    alias AntikytheraCore.GearLog.Writer, warn: false
```

Reply:
Thank you for pointing out.
I missed the warning produced by compiling from scratch.

I also incorporated your suggestion that `Mix.env() == :test` is more intuitive.

Reply:
> `if Antikythera.Env.compile_env() == :undefined do`

Sorry, I did not explain why I wrote this.
Because here should be `true` when `mix test` is performed in a gear project (so antikythera is in dependencies and `Mix.env()` here becomes `:prod`).

I've reverted to the old code for this line. (https://github.com/access-company/antikythera/pull/182/commits/cc7f84cbfe0c4306c8cc3b9393d0eaa81c26086c)